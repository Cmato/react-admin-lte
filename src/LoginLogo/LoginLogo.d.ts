import * as React from 'react';

export interface LoginLogoProps {
}

declare const LoginLogo: React.ComponentType<LoginLogoProps>;

export default LoginLogo;
