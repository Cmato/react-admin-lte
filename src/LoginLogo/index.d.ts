export { default } from './LoginLogo';
export * from './LoginLogo';
