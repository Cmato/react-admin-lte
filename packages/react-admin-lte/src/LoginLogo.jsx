import PropTypes from 'prop-types';
import React from 'react';
import classNames from 'classnames';

const propTypes = {
  children: PropTypes.node,
  className: PropTypes.node,
};

const LoginLogo = ({ children, className }) => {
  const classes = {
    "login-logo": true,
  };

  return (
    <div className={classNames(className, classes)}>
      {children}
    </div>
  );
};

LoginLogo.propTypes = propTypes;

export default LoginLogo;
