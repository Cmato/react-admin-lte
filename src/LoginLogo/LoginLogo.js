import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

const LoginLogo = ({
  children,
  className: classNameProp,
  ...other
}) => {
  const className = classNames({
    'login-logo': true
  }, classNameProp);

  return (
    <div className={className} {...other}>
      {children}
    </div>
  );
};

LoginLogo.propTypes = {
  /**
   * The content of the component.
   */
  children: PropTypes.node.isRequired,
  /**
   * @ignore
   */
  className: PropTypes.string,
};

export default LoginLogo;
