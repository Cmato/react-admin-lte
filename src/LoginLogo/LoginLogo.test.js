import React from 'react';
import { shallow } from 'enzyme';
import LoginLogo from './LoginLogo';

it('renders default', () => {
  expect(shallow(<LoginLogo>Mine!</LoginLogo>))
    .toMatchSnapshot();
});
