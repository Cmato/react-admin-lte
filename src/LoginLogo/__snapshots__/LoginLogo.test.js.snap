// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders default 1`] = `
<div
  className="login-logo"
>
  Mine!
</div>
`;
